powersimdata/scenario/create.py

```diff
@@ -344,31 +415,52 @@ def set_base_profile(self, kind, version):
         else:
             raise ValueError("Available %s profiles: %s" % (kind, " | ".join(possible)))
 
-    def set_engine(self, engine):
-        """Sets simulation engine to be used for scenarion.
 
-        :param str engine: simulation engine
-        """
-        possible = ["REISE.jl"]
-        if engine not in possible:
-            print("Available engines: %s" % " | ".join(possible))
-            return
-        else:
-            self.engine = engine
+class FromPyPSA(_Builder):
+    """Build scenario from a PyPSA Network object
 
-    def get_grid(self):
-        """Returns a transformed grid.
+    :param str grid model: the grid model
+    :param list interconnect: list of interconnect(s) to build.
+    :param pandas.DataFrame table: scenario list table.
+    :param \\*\\*kwargs: optional parameters used to instantiate a Grid object:
+        *'reduction'*: number of nodes in the network. If None, the full resolution
+        PyPSA Network object will be used. Available reductions are specified in the
+        :mod:`powersimdata.network.europe_tub.model` module.
+    """
 
-        :return: (*powersimdata.input.grid.Grid*) -- a Grid object.
-        """
-        return TransformGrid(self.base_grid, self.change_table.ct).get_grid()
+    def __init__(self, grid_model, interconnect, table, **kwargs):
+        super().__init__(grid_model, interconnect, table)
 
-    def get_base_grid(self):
-        """Returns original grid.
+        self.reduction = None if "reduction" not in kwargs else kwargs["reduction"]
 
-        :return: (*powersimdata.input.grid.Grid*) -- a Grid object.
-        """
-        return copy.deepcopy(self.base_grid)
+        self.print_existing_study()
 
-    def __str__(self):
-        return self.name
+        self.set_base_grid()
+        self.set_change_table()
+
+    def set_base_grid(self):
+        """Set base grid"""
+        raise NotImplementedError()
+
+    def set_change_table(self):
+        """Set change table"""
+        raise NotImplementedError()
+
+
+class BuilderFactory:
+    """Map a grid model to a builder"""
+
+    MAP = {"usa_tamu": FromCSV, "hifld": FromCSV, "europe_tub": FromPyPSA}
```

Comment:
I'm trying to understand why we will need a class to handle this static mapping. Do we have plans to deal with more complicated situations here in the future?

Reply:
This is usually how this is done, e.g., see the [`config`](https://github.com/Breakthrough-Energy/PowerSimData/blob/develop/powersimdata/utility/config.py) module. I agree that the mapping could be moved in the `get_builder` function. @jenhagg, what is your take on that?

Reply:
I agree, we don't need a class here, just having the mapping in `get_builder` is fine. I'm not sure we even need the one in the config module. I think it was done that way to have a single place for the string constants but we have e.g. `"usa_tamu"` in lots of places so that wouldn't really accomplish much in this case.

Reply:
Done